### PR TITLE
Add Texture2D.SetDataPointerEXT like on the VertexBuffer

### PR DESCRIPTION
**Pull request by bartwe**

### CODE REVIEW

File: src/Graphics/Texture2D.cs [hunk 1]
```diff
@@ -158,7 +158,7 @@ int elementCount
 				w = Math.Max(Width >> level, 1);
 				h = Math.Max(Height >> level, 1);
 			}
-
+			var elementSize = Marshal.SizeOf(typeof(T));
```
Comment (flibitijibibo) [inline src/Graphics/Texture2D.cs]: Only one `var`, too! Aside from this the patch is good, so just rinse and repeat for Get and 3D/Cube.

**Comment by bartwe** [comment]
Added the additional changes and removed the 'var'

Could not change IGLDevice due to the implementations requiring these fields for the other functions which also leaves the SetDataPointerEXT  more verbose in those cases

**Comment by flibitijibibo** [comment]
For GetData it looks like you're right; even for buffer objects I opted to stick with the element stuff... SetData, however, I'm not seeing what'd make it more verbose. The GL impls wouldn't use `dataLength`, for example, seeing that they don't use `elementCount`, but VulkanDevice will need it for the memcpy operation and the pointer offsets that are calculated in the Devices can be done by the caller.

So I guess we can leave GetData alone (so that extension can be thrown out, sorry about that!) but SetDataPointer can definitely be applied to all three types.

**Comment by bartwe** [comment]
Ok so what do you want me to do with the PR ? remove the GetData ones  and
further nothing ?

File: src/Graphics/Texture3D.cs [hunk 2]
```diff
@@ -129,14 +130,42 @@ int elementCount
...
+			int left,
+			int top,
+			int right,
+			int bottom,
+			int front,
+			int back,
+			IntPtr data,
+			int dataLength
+		) {
+			if (data == null) {
```
Comment (flibitijibibo) [inline src/Graphics/Texture3D.cs]: Just this one brace, otherwise this is now set to go!

**Comment by bartwe** [comment]
My autoformatter hates your autoformatter ;) do you maybe just have a
resharper autoformatter template i can import so it does this automatically
instead of manually ?

**Comment by flibitijibibo** [comment]
I don't have any autoformatting locally, just autotabbing, so unfortunately I don't have any templates for anything.

**Comment by bartwe** [comment]
pushed whitespace change

**Comment by flibitijibibo** [comment]
[Docs have been updated.](https://github.com/FNA-XNA/FNA/wiki/5:-FNA-Extensions#setdatapointerext)

**Pull request merged**